linkerd/app/integration/src/tests/telemetry/tls_detect.rs

```diff
@@ -0,0 +1,124 @@
+use super::*;
+use std::time::SystemTime;
+
+const TIMEOUT: Duration = Duration::from_millis(640); // 640ms ought to be enough for anybody.
+
+/// A helper that builds a proxy with the above detect timeout and a TCP server that always drops
+/// the accepted socket.
+async fn run_proxy() -> proxy::Listening {
+    let identity::Identity {
+        mut env,
+        mut certify_rsp,
+        ..
+    } = identity::Identity::new(
+        "foo-ns1",
+        "foo.ns1.serviceaccount.identity.linkerd.cluster.local".to_string(),
+    );
+
+    // The identity service is needed for the proxy to start.
+    let id_svc = {
+        certify_rsp.valid_until = Some((SystemTime::now() + Duration::from_secs(666)).into());
+        controller::identity()
+            .certify(move |_| certify_rsp)
+            .run()
+            .await
+    };
+
+    // We provide a mocked TCP server that always immediately drops accepted socket. This should
+    // trigger errors.
+    let srv = tcp::server()
+        .accept_fut(move |sock| async { drop(sock) })
+        .run()
+        .await;
```

Comment:
I don't actually understand how this works for triggering timeouts...

Reply:
it doesn't, we don't actually talk to the server. it's just there so that there's an ORIG_DST port.